### PR TITLE
Buff dimensional tears.

### PR DESCRIPTION
**Pull request by warriorstar-orion**
## What Does This PR Do
This PR makes several changes to dimensional tears/honknomolies. Before this PR, the way tears work is:

1. Spook the machines/lights in a radius around the spawn point.
2. Shortly after, break nearby computers.
3. Shortly after, spawn in 5 mobs randomly chosen from a list of (migo, faithless, "creature" (pink big mouth dudes)).
4. Shortly after, spawn in a "leader" mob (which is a hellhound that has been uniquely buffed for some reason)

After this PR:

1. Spook the machines/lights.
2. Shortly after, break nearby computers _and_ lights.
3. Spawn one mob from a list of (migo, faithless, creature, skeleton, hellhound) every two seconds to a maximum of d6+4 mobs.
4. Spawn the hellhound and remove the portal.

For honknomolies:

1. Honknomolies only had one enemy in their mob list, clown goblins, and only had a 5% chance of spawning a leader mob, which is a cluwne goblin. After this PR, cluwne goblins, as well as hostile clown simplemobs, are now just potential mobs to spawn.

Additionally:

1. Tears will no longer spawn on walls and force mobs e.g. into space when spawning.
2. A message will be logged if the tear cannot find an empty turf to spawn onto.
3. DMdocs have been added.

## Why It's Good For The Game
Dimensional tears/honknomolies are categorized as "moderate" events, which puts them in the same category as wormholes, vortex anomalies, immovable rods, and meteor showers. They should provide much more bang for their buck and some manner of risk. In my ideal world tears would be as bad as BSH incursions, and incursions would be much much much much worse but this is a good start.

## Images of changes
No real visual changes.

## Testing
Triggered event multiple times.

## Changelog
:cl:
tweak: dimensional tears and honknomolies are relatively more challenging
/:cl:

### CODE REVIEW

**Comment by warriorstar-orion** [comment]
Todo from discord discussion: bring back tear hellhound and guarantee spawn

File: code/modules/events/tear.dm [hunk 1]
```diff
@@ -18,8 +18,10 @@
 	impact_area = findEventArea()
 
 /datum/event/tear/start()
-	var/turf/T = pick(get_area_turfs(impact_area))
-	if(T)
+	for(var/turf/T in get_area_turfs(impact_area))
+		if(is_blocked_turf(T))
+			continue
```
Comment (Contrabang) [inline code/modules/events/tear.dm:23]: Doesn't this mean it will always pick the same turf? Maybe a shuffling of the list first is needed?

Comment (warriorstar-orion) [reply code/modules/events/tear.dm:23]: do you know offhand if there's a helper proc to do this? i feel like getting a random non-dense tile in a given area is a pretty common need but i couldn't find any utility methods to do it.

Comment (warriorstar-orion) [reply code/modules/events/tear.dm]: PTAL

File: code/modules/events/tear.dm [hunk 2]
```diff
@@ -18,8 +18,12 @@
 	impact_area = findEventArea()
 
 /datum/event/tear/start()
-	var/turf/T = pick(get_area_turfs(impact_area))
-	if(T)
+	var/list/area_turfs = get_area_turfs(impact_area)
+	while(length(area_turfs) > 0)
```
Comment (S34NW) [inline code/modules/events/tear.dm:22]:
```suggestion
	while(length(area_turfs))
```
surely

**Review by S34NW — CHANGES_REQUESTED** [review]

**Review by Contrabang — APPROVED** [review]
LGTM (excluding s34n's review) and except the `The custom hellhound subtype used exclusively for the tear event has been removed.` in the description of the PR, because I believe that was readded. But that's relatively minor.

File: code/modules/events/tear.dm [hunk 3]
```diff
@@ -29,6 +33,10 @@
 		for(var/obj/machinery/M in range(8, T))
 			INVOKE_ASYNC(M, TYPE_PROC_REF(/atom, get_spooked))
 
+		return
+
+	log_and_message_admins("dimensional tear failed to find a valid turf in [impact_area]")
```
Comment (lewcc) [inline code/modules/events/tear.dm:38]: most events make use of log_debug, could be good to include such a call here

Comment (Contrabang) [reply code/modules/events/tear.dm:38]: Agreed, also `log_and_message_admins` uses keyname usr, which is probably going to break here

**Comment by warriorstar-orion** [comment]
> LGTM (excluding s34n's review) and except the `The custom hellhound subtype used exclusively for the tear event has been removed.` in the description of the PR, because I believe that was readded. But that's relatively minor.

Updated the description

File: code/modules/events/tear.dm (hunk 3 above)
Comment (warriorstar-orion) [reply code/modules/events/tear.dm]: switched to log_debug here!

**Review by DGamerL — APPROVED** [review]
LGTM

**Review by S34NW — APPROVED** [review]

**Pull request merged**